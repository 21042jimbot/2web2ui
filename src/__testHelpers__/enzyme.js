export const findButton = (wrapper, buttonLabel) =>
  wrapper.findWhere(
    (node) => node.type() === 'button' && new RegExp(buttonLabel).test(node.text())
  );
