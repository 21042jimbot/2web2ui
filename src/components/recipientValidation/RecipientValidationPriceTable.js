import React from 'react';
import { Table } from '@sparkpost/matchbox';
import { RECIPIENT_VALIDATION_TIERS } from 'src/constants';
import { formatFullNumber } from 'src/helpers/units';
import styles from './RecipientValidationPriceTable.module.scss';

export const RecipientValidationPriceTable = ({ cellProps, col1Props, col2Props }) => (
  <Table>
    <tbody>
      <Table.Row className={styles.tableHeader}>
        <Table.Cell {...cellProps} {...col1Props}>
          <strong>Number of Emails</strong>
        </Table.Cell>
        <Table.Cell {...cellProps} {...col2Props}>
          <strong>Cost</strong>
        </Table.Cell>
      </Table.Row>
      {RECIPIENT_VALIDATION_TIERS.map(row => (
        <Table.Row key={`row-${row.volumeMin}-${row.volumeMax}`}>
          <Table.Cell {...cellProps} {...col1Props}>
            <strong>{formatFullNumber(row.volumeMin)}</strong>
            {row.volumeMax < Infinity ? ( //Last row with volumeMax of Infinity has different wording
              <>
                {' '}
                to <strong>{formatFullNumber(row.volumeMax)}</strong>
              </>
            ) : (
              <>+</>
            )}
          </Table.Cell>
          <Table.Cell {...cellProps} {...col2Props}>
            <span>
              <strong>{row.displayedCost}</strong> per email address
            </span>
          </Table.Cell>
        </Table.Row>
      ))}
    </tbody>
  </Table>
);

export default RecipientValidationPriceTable;
