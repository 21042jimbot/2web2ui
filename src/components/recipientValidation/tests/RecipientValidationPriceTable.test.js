import React from 'react';
import { shallow } from 'enzyme';
import RecipientValidationPriceTable from '../RecipientValidationPriceTable';

describe('RecipientValidationTable', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<RecipientValidationPriceTable />);
    expect(wrapper).toMatchSnapshot();
  });
});
