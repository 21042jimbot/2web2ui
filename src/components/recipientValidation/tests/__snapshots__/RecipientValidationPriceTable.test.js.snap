// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipientValidationTable should render correctly 1`] = `
<Table>
  <tbody>
    <Table.Row
      className="tableHeader"
    >
      <Table.Cell>
        <strong>
          Number of Emails
        </strong>
      </Table.Cell>
      <Table.Cell>
        <strong>
          Cost
        </strong>
      </Table.Cell>
    </Table.Row>
    <Table.Row
      key="row-0-5000"
    >
      <Table.Cell>
        <strong>
          0
        </strong>
         
        to 
        <strong>
          5,000
        </strong>
      </Table.Cell>
      <Table.Cell>
        <span>
          <strong>
            $0.010
          </strong>
           per email address
        </span>
      </Table.Cell>
    </Table.Row>
    <Table.Row
      key="row-5000-10000"
    >
      <Table.Cell>
        <strong>
          5,000
        </strong>
         
        to 
        <strong>
          10,000
        </strong>
      </Table.Cell>
      <Table.Cell>
        <span>
          <strong>
            $0.008
          </strong>
           per email address
        </span>
      </Table.Cell>
    </Table.Row>
    <Table.Row
      key="row-10000-50000"
    >
      <Table.Cell>
        <strong>
          10,000
        </strong>
         
        to 
        <strong>
          50,000
        </strong>
      </Table.Cell>
      <Table.Cell>
        <span>
          <strong>
            $0.006
          </strong>
           per email address
        </span>
      </Table.Cell>
    </Table.Row>
    <Table.Row
      key="row-50000-100000"
    >
      <Table.Cell>
        <strong>
          50,000
        </strong>
         
        to 
        <strong>
          100,000
        </strong>
      </Table.Cell>
      <Table.Cell>
        <span>
          <strong>
            $0.004
          </strong>
           per email address
        </span>
      </Table.Cell>
    </Table.Row>
    <Table.Row
      key="row-100000-250000"
    >
      <Table.Cell>
        <strong>
          100,000
        </strong>
         
        to 
        <strong>
          250,000
        </strong>
      </Table.Cell>
      <Table.Cell>
        <span>
          <strong>
            $0.003
          </strong>
           per email address
        </span>
      </Table.Cell>
    </Table.Row>
    <Table.Row
      key="row-250000-750000"
    >
      <Table.Cell>
        <strong>
          250,000
        </strong>
         
        to 
        <strong>
          750,000
        </strong>
      </Table.Cell>
      <Table.Cell>
        <span>
          <strong>
            $0.0015
          </strong>
           per email address
        </span>
      </Table.Cell>
    </Table.Row>
    <Table.Row
      key="row-750000-1000000"
    >
      <Table.Cell>
        <strong>
          750,000
        </strong>
         
        to 
        <strong>
          1,000,000
        </strong>
      </Table.Cell>
      <Table.Cell>
        <span>
          <strong>
            $0.0010
          </strong>
           per email address
        </span>
      </Table.Cell>
    </Table.Row>
    <Table.Row
      key="row-1000000-Infinity"
    >
      <Table.Cell>
        <strong>
          1,000,000
        </strong>
        +
      </Table.Cell>
      <Table.Cell>
        <span>
          <strong>
            $0.00075
          </strong>
           per email address
        </span>
      </Table.Cell>
    </Table.Row>
  </tbody>
</Table>
`;
